vllm/model_executor/guided_decoding/outlines_logits_processors.py

```diff
@@ -30,11 +30,48 @@
 from transformers import PreTrainedTokenizerBase
 
 
+# Unfortunately we cannot use lru_cache as it breaks pickling
+# so we use a simpler implementation
+def _cached(fn):
```

Comment:
How many masks are expected at maximu? Since _create_mask_tensor is limited and the _cached fun is not, won't we get into a situation where we will get None result for least recently used (128+)?

Reply:
You can ignore my last comment, the lrucache maxsize does not have effect here, since _cached cache keeps objects.
If you reverse the condition in line 39 the else can be replaced with return cache[args]
         if args not in cache:
            cache[args] = fn(*args)
        return cache[args]

Reply:
Yeah, but at the same time we'd need to calculate the hash and access the map an additional time.
1) if args in cache
2) cache[args] = fn(*args)
3) return cache[args]

In current version you access it twice:
1) args in cache
2) cache[args] = result

To be honest I didn't measure how big of a perf impact this might be, but I'd like to be on the safe side.